crates/next-core/src/middleware.rs

```diff
@@ -57,3 +116,51 @@ pub async fn get_middleware_module(
 
     Ok(module)
 }
+
+#[turbo_tasks::value]
+struct MiddlewareMissingExportIssue {
+    file_type: RcStr,     // "Proxy" or "Middleware"
+    function_name: RcStr, // "proxy" or "middleware"
+    file_path: FileSystemPath,
+}
+
+#[turbo_tasks::value_impl]
+impl Issue for MiddlewareMissingExportIssue {
+    #[turbo_tasks::function]
+    fn stage(&self) -> Vc<IssueStage> {
+        IssueStage::Transform.into()
+    }
+
+    fn severity(&self) -> IssueSeverity {
+        IssueSeverity::Error
+    }
+
+    #[turbo_tasks::function]
+    fn file_path(&self) -> Vc<FileSystemPath> {
+        self.file_path.clone().cell()
+    }
+
+    #[turbo_tasks::function]
+    fn title(&self) -> Vc<StyledString> {
+        let file_name = Path::new(&self.file_path.path)
+            .file_name()
+            .and_then(|n| n.to_str())
+            .unwrap_or(&self.file_path.path);
+
+        StyledString::Line(vec![
+            StyledString::Text(rcstr!("The ")),
+            StyledString::Code(self.file_type.clone()),
+            StyledString::Text(rcstr!(" file \"")),
+            StyledString::Code(format!("./{}", file_name).into()),
+            StyledString::Text(rcstr!("\" must export a function named ")),
+            StyledString::Code(format!("`{}`", self.function_name).into()),
```

Comment:
Nit: Leave the formatting choice of backticks around code to the thing doing the formatting.

```suggestion
            StyledString::Code(self.function_name.clone()),
```